v3/cmd/zlint/main.go

```diff
@@ -203,6 +216,11 @@ func trimmedList(raw string) []string {
 // use.
 //nolint:cyclop
 func setLints() (lint.Registry, error) {
+	config, err := lint.NewConfigFromFile(config)
```

Comment:
nit: this kind of shadowing (here, `config` is now of type `Configuration`, instead of type `string`) can be difficult to read. I'd use a different variable name here to avoid any chance of confusion.